src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,183 @@
+                {
+                    if (ex is SolutionException || ex.InnerException is SolutionException)
+                    {
+                        throw new GracefulException(CommonLocalizableStrings.InvalidSolutionFormatString, solutionFileFullPath, ex.Message);
+                    }
+                    throw new GracefulException(ex.Message, ex);
+                }
             }
-
-            return 0;
         }
 
-        private static IList<string> GetSolutionFoldersFromProjectPath(string projectFilePath)
+        private async Task AddProjectsToSolutionAsync(string solutionFileFullPath, IEnumerable<string> projectPaths, CancellationToken cancellationToken)
         {
-            var solutionFolders = new List<string>();
-
-            if (!IsPathInTreeRootedAtSolutionDirectory(projectFilePath))
-                return solutionFolders;
-
-            var currentDirString = $".{Path.DirectorySeparatorChar}";
-            if (projectFilePath.StartsWith(currentDirString))
+            ISolutionSerializer serializer = SlnCommandParser.GetSolutionSerializer(solutionFileFullPath);
+            SolutionModel solution = await serializer.OpenAsync(solutionFileFullPath, cancellationToken);
+            // set UTF8 BOM encoding for .sln
+            if (serializer is ISolutionSerializer<SlnV12SerializerSettings> v12Serializer)
             {
-                projectFilePath = projectFilePath.Substring(currentDirString.Length);
+                solution.SerializerExtension = v12Serializer.CreateModelExtension(new()
+                {
+                    Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true)
+                });
             }
-
-            var projectDirectoryPath = TrimProject(projectFilePath);
-            if (string.IsNullOrEmpty(projectDirectoryPath))
-                return solutionFolders;
-
-            var solutionFoldersPath = TrimProjectDirectory(projectDirectoryPath);
-            if (string.IsNullOrEmpty(solutionFoldersPath))
-                return solutionFolders;
-
-            solutionFolders.AddRange(solutionFoldersPath.Split(Path.DirectorySeparatorChar));
-
-            return solutionFolders;
-        }
-
-        private IList<string> DetermineSolutionFolder(SlnFile slnFile, string fullProjectPath)
-        {
-            if (_inRoot)
+            // Set default configurations and platforms for sln file
+            foreach (var platform in new[]{ "Any CPU", "x64", "x86" })
             {
-                // The user requested all projects go to the root folder
-                return null;
+                solution.AddPlatform(platform);
             }
-
-            if (_relativeRootSolutionFolders != null)
+            foreach (var buildType in new []{ "Debug", "Release" })
             {
-                // The user has specified an explicit root
-                return _relativeRootSolutionFolders;
+                solution.AddBuildType(buildType);
```

Comment:
Should we be detecting what the solution currently does for any of these? I'm wondering if it could turn, for instance, a debug-only slnx into debug|release

Reply:
I believe it mainly serves for mapping with the projects' build type